doc/GUIDE.md

```diff
@@ -0,0 +1,848 @@
+    Foo.bar.baz(bar2, 42);
+    Foo.bar(foo, bar2);  // copies bar2 into foo
+    System.out.println(Foo.bar.baz(bar));  // prints: 42
+}
+```
+
+In the above snippet, note that the load of the `baz` field value on the last line will
+_see_ the update to the `bar` field of the `foo` instance on the line before.
+
+## Filtering
+
+Some libraries are incredibly large (such as `Windows.h`), and we might not be
+interested in letting jextract generate code for the entire library. In cases like that,
+we can use jextract's `--include-XXX` command line options to only generate classes for
+the elements we specify.
+
+To allow for symbol filtering, jextract can generate a _dump_ of all the symbols
+encountered in an header file; this dump can be manipulated, and then used as an argument
+file (using the `@argfile` syntax also available in other JDK tools) to e.g. generate
+bindings only for a _subset_ of symbols seen by jextract. For instance, if we run
+jextract with as follows:
+
+```sh
+$ jextract --dump-includes includes.txt mylib.h
+```
+
+We obtain the following file (`includes.txt`):
+
+```sh
+#### Extracted from: /workspace/myproj/mylib.h
+
+--include-struct Foo                  # header: /workspace/myproj/mylib.h
+--include-struct Point                # header: /workspace/myproj/mylib.h
+--include-typedef callback_t          # header: /workspace/myproj/mylib.h
+--include-function call_me_back       # header: /workspace/myproj/mylib.h
+...
+```
+
+The include options in this file can then be edited down to a set of symbols that is
+desired, for instance using other command line tools such as `grep` or `Select-String`,
+and passed back to jextract:
+
+```sh
+$ jextract --dump-includes includes.txt mylib.h
+$ grep Foo includes.txt > includes_filtered.txt
+$ jextract @includes_filtered.txt mylib.h
+```
+
+Users should exercise caution when filtering symbols, as it is relatively easy to filter
+out a declaration that is depended on by one or more declarations:
+
+```c
+// test.h
+struct A {
+   int x;
+}
+struct A aVar;
+```
+
+Here, we could run jextract and filter out `A`, like so:
+
+```sh
+$ jextract --include-var aVar test.h
+```
+
+However, doing so would lead to broken generated code, as the layout of the global variable
+`aVar` depends on the layout of the excluded struct `A`.
+
+In such cases, jextract will report the missing dependency and terminate without
+generating any bindings:
+
+```txt
+ERROR: aVar depends on A which has been excluded
+```
+
+## Tracing
+
+It is sometimes useful to inspect the parameters passed to a native call, especially when
+diagnosing application bugs and/or crashes. The code generated by the jextract tool
+supports _tracing_ of native calls, that is, parameters passed to native calls can be
+printed on the standard output.
+
+To enable the tracing support, just pass the `-Djextract.trace.downcalls=true` flag to the
+launcher used to start the application. Below we show an excerpt of the output when
```

Comment:
I would mention that this is passed as a VM argument, not as a program argument.